modules/googleplay/src/main/scala/cards/nine/googleplay/service/free/interpreter/cache/CacheInterpreter.scala

```diff
@@ -37,48 +36,52 @@ class CacheInterpreter(implicit ec: ExecutionContext) extends (Ops ~> RedisOps)
-      val keys = (packages map CacheKey.resolved) ++ (packages map CacheKey.permanent)
-      wrap.mget(keys).map(_.flatMap(_.card))
+  override def getValid(pack: Package): RedisOps[Option[FullCard]] = {
+    val keys = List(CacheKey.resolved(pack), CacheKey.permanent(pack))
+    wrap.mget(keys).map(_.flatMap(_.card).headOption)
+  }
 
-    case PutResolved(card) ⇒
-      val pack = card.packageName
-      pendingSet.remove(pack) *> removeError(pack) *> wrap.put(CacheEntry.resolved(card))
+  override def getValidMany(packages: List[Package]) = {
+    val keys = (packages map CacheKey.resolved) ++ (packages map CacheKey.permanent)
+    wrap.mget(keys).map(_.flatMap(_.card))
+  }
 
-    case PutResolvedMany(cards) ⇒
-      val packs = cards.map(_.packageName)
-      wrap.mput(cards map CacheEntry.resolved) *> removeErrorMany(packs) *> pendingSet.remove(packs)
+  override def putResolved(card: FullCard): RedisOps[Unit] = {
+    val pack = card.packageName
+    pendingSet.remove(pack) *> removeError(pack) *> wrap.put(CacheEntry.resolved(card))
+  }
 
-    case PutPermanent(card) ⇒
-      val pack = card.packageName
-      pendingSet.remove(pack) *> removeError(pack) *> wrap.put(CacheEntry.permanent(card))
+  override def putResolvedMany(cards: List[FullCard]): RedisOps[Unit] = {
+    val packs = cards.map(_.packageName)
+    wrap.mput(cards map CacheEntry.resolved) *> removeErrorMany(packs) *> pendingSet.remove(packs)
+  }
 
-    case SetToPending(pack) ⇒
-      removeError(pack) *> pendingSet.insert(pack)
+  override def putPermanent(card: FullCard): RedisOps[Unit] = {
+    val pack = card.packageName
+    pendingSet.remove(pack) *> removeError(pack) *> wrap.put(CacheEntry.permanent(card))
+  }
 
-    case SetToPendingMany(packages) ⇒
-      removeErrorMany(packages) *> pendingSet.insert(packages)
+  override def setToPending(pack: Package): RedisOps[Unit] =
+    removeError(pack) *> pendingSet.insert(pack)
 
-    case AddError(pack) ⇒
-      val now = DateTime.now(DateTimeZone.UTC)
-      pendingSet.remove(pack) *> errorCache.enqueue(CacheKey.error(pack), now)
+  override def setToPendingMany(packages: List[Package]): RedisOps[Unit] =
+    removeErrorMany(packages) *> pendingSet.insert(packages)
 
-    case AddErrorMany(packages) ⇒
-      val now = DateTime.now(DateTimeZone.UTC)
-      val putErrors = packages.traverse[RedisOps, Unit] {
-        pack ⇒ errorCache.enqueue(CacheKey.error(pack), now)
-      }
-      putErrors *> pendingSet.remove(packages)
+  override def addError(pack: Package): RedisOps[Unit] = {
+    val now = DateTime.now(DateTimeZone.UTC)
+    pendingSet.remove(pack) *> errorCache.enqueue(CacheKey.error(pack), now)
+  }
 
-    case ListPending(num) ⇒ pendingSet.extractMany(num)
+  override def addErrorMany(packages: List[Package]): RedisOps[Unit] = {
+    val now = DateTime.now(DateTimeZone.UTC)
+    val putErrors = packages.traverse[RedisOps, Unit] {
+      pack ⇒ errorCache.enqueue(CacheKey.error(pack), now)
+    }
```

Comment:
Again unrelated: Since we are returning `Unit`, we can replace `traverse` here with `traverse_` (from `Foldable`).